fairlearn/metrics/_metric_frame.py

```diff
@@ -372,29 +379,67 @@ def sensitive_levels(self) -> List[str]:
         """
         return self._sf_names
 
-    def group_max(self) -> Union[Any, pd.Series, pd.DataFrame]:
-        """Return the maximum value of the metric over the sensitive features.
+    def group(self, grouping_function: str, errors: str = 'raise') -> Union[Any, pd.Series, pd.DataFrame]:
```

Comment:
This should probably be `_group()` and so not public (well, by convention).

Reply:
Ah yes makes sense! It should be a private method.